core/src/wallet/bitcoin/synchronizers/BlockchainExplorerAccountSynchronizer.cpp

```diff
@@ -95,5 +135,191 @@ namespace ledger {
+            .flatMap<TransactionBatchVector>(buddy->account->getContext(), [=] (const auto& block) {
+                std::vector<Future<std::vector<Transaction>>> getTxs;
+                for(std::size_t i = 0; i < addresses.size(); i += ADDRESS_BATCH_SIZE) {
+                    auto last = std::min(addresses.size(), i + ADDRESS_BATCH_SIZE);
+                    std::vector<std::string> batch(addresses.begin() + i, addresses.begin() + last);
+                    getTxs.emplace_back(getMempoolTransactionBatch(buddy, block, batch));
+                }
+                return async::sequence(buddy->account->getContext(), getTxs);
+            }).map<std::vector<Transaction>>(buddy->account->getContext(), [] (const auto& batches) {
+                std::vector<Transaction> mempoolTxs;
+                for (const auto& batch : batches) {
+                    mempoolTxs.insert(mempoolTxs.end(), batch.begin(), batch.end());
+                }
+                return mempoolTxs;
+            });
+        }
+
+        Future<Unit> resolveMempool(const std::shared_ptr<BitcoinSynchronizationBuddy>& buddy) {
+            // Do a first round of filtering with transaction bound to the account
+            TransactionMap txsByHash;
+            filterReplacedTransaction(buddy->mempoolTransaction, txsByHash);
+            // Get all transaction which are still replaceable and create a list of addresses
+            std::list<Transaction> notReplaceableTxs;
+            std::list<Transaction> replaceableTxs;
+            std::unordered_set<std::string> foreignAddresses;
+
+            for (const auto& txWithHash : txsByHash) {
+                if (txWithHash.second.second) {
+                    // Tx is replaceable add it in the list to filter later
+                    // and go though all inputs to find foreign address
+                    replaceableTxs.emplace_back(txWithHash.second.first);
+                    for (const auto& input : txWithHash.second.first.inputs) {
+                        if (input.address.hasValue() &&
+                            !buddy->account->getKeychain()->contains(input.address.getValue())) {
+                            foreignAddresses.insert(input.address.getValue());
+                        }
+                    }
+                } else {
+                    notReplaceableTxs.emplace_back(txWithHash.second.first);
+                }
+            }
+            // Use the list of addresses to get all mempool transaction for those addresses
+            return getMempoolTransactions(buddy,
+                    std::vector<std::string>(foreignAddresses.begin(), foreignAddresses.end()))
+            .map<Unit>(buddy->account->getContext(), [=] (const auto& foreignTxs) {
+                // Do another round of filtering on replaceable transaction vs foreign transactions
+                TransactionMap filteredTxs;
+                std::list<Transaction> txs(replaceableTxs.begin(), replaceableTxs.end());
```

Comment:
Why use `std::list` ?

Reply:
no valuable reason, I'll change it to vector.